guides/plugins/plugins/storefront/add-scss-variables.md

```diff
@@ -0,0 +1,212 @@
+Please note that plugins are not sales-channel specific. Your SCSS variables are directly added in the SCSS compilation process and will be globally available throughout all themes and storefront sales-channels. If you want to change a variables value for each sales-channel you should use plugin config fields and follow the next example.
+{% endhint %}
+
+## Plugin config values as SCSS variables
+
+Inside your `ThemeVariablesSubscriber.php` you can also read values from the plugin configuration and assign those to a SCSS variable. This makes it also possible to have different values for each sales-channel. Depending on the selected sales-channel inside the plugin configuration in the administration.
+
+First of all lets add a new plugin configuration field according to the [Plugin Configurations](../plugin-fundamentals/add-plugin-configuration.md):
+
+{% code title="<plugin root>/src/Resources/config/config.xml" %}
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/shopware/platform/trunk/src/Core/System/SystemConfig/Schema/config.xsd">
+
+    <card>
+        <title>Example configuration</title>
+        <input-field>
+            <name>sassPluginHeaderBgColor</name>
+            <label>Header background color</label>
+        </input-field>
+    </card>
+</config>
+```
+{% endcode %}
+
+To be able to read this config you have to add the `SystemConfigService` to your subscriber:
+
+{% tabs %}
+{% tab title="<plugin root>/src/Subscriber/ThemeVariablesSubscriber.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Subscriber;
+
+use Shopware\Core\System\SystemConfig\SystemConfigService;
+use Shopware\Storefront\Event\ThemeCompilerEnrichScssVariablesEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class ThemeVariablesSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var SystemConfigService
+     */
+    protected $systemConfig;
+
+    // add the `SystemConfigService` to your constructor
+    public function __construct(SystemConfigService $systemConfig)
+    {
+        $this->systemConfig = $systemConfig;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            ThemeCompilerEnrichScssVariablesEvent::class => 'onAddVariables'
+        ];
+    }
+
+    public function onAddVariables(ThemeCompilerEnrichScssVariablesEvent $event)
+    {
+        /** @var string $configExampleField */
+        $configPluginHeaderBgColor = $this->systemConfig->get('SwagBasicExample.config.sassPluginHeaderBgColor', $event->getSalesChannelId());
+
+        // pass the value from `configPluginHeaderBgColor` to `addVariable`
+        $event->addVariable('sass-plugin-header-bg-color', $configPluginHeaderBgColor);
+    }
+}
+```
+{% endtab %}
+{% tab title="<plugin root>/src/Resources/config/services.xml" %}
```

Comment:
I like that tab style btw. for those examples 👍

Reply:
Will this end up in rewriting every code-block into a tab-block? Will this remove feature that the code-block can provide in comparison to the tab-block?

Reply:
@JoshuaBehrens - I don't think it should be used everywhere. In this guide I thought it would be better to keep the code samples for the subscriber and the `service.xml` entry together.